src/Npgsql.EntityFrameworkCore.PostgreSQL/Migrations/Internal/NpgsqlHistoryRepository.cs

```diff
@@ -85,22 +85,31 @@ protected override string ExistsSql
 
         public override string GetCreateIfNotExistsScript()
         {
-            return GetCreateScript();
+            var script = GetCreateScript();
+            return script.Insert(script.IndexOf("CREATE TABLE", StringComparison.Ordinal) + 12, " IF NOT EXISTS");
         }
 
         public override string GetBeginIfNotExistsScript(string migrationId)
         {
-            throw new NotSupportedException("Generating idempotent scripts for migration is not currently supported by Npgsql");
+            return $@"
+DO $$
+BEGIN
+    IF NOT EXISTS(SELECT 1 FROM {this.TableSchema}.""{this.TableName}"" WHERE ""{this.MigrationIdColumnName}"" = '{migrationId}') THEN";
```

Comment:
`this` here is unneeded (I know I'm nitpicking, just trying to get the code to use the same style rules)